rules/prefer-type-error.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+const tcIdentifiers = new Set([
+	'isArguments',
+	'isArray',
+	'isArrayBuffer',
+	'isArrayLike',
+	'isArrayLikeObject',
+	'isBoolean',
+	'isBuffer',
+	'isDate',
+	'isElement',
+	'isEmptyObject',
+	'isError',
+	'isFinite',
```

Comment:
Allright, let's keep it this way then. We can always reconsider if someone has issues with it later on.